R/estimate_dispersion.R

```diff
@@ -74,9 +74,21 @@ estimate_dispersion <- function(
 
   assert_integerish(n, lower = 0)
 
-  # Truncate to only n nowcasts
-  list_of_ncs <- pt_nowcast_mat_list[1:n]
-  list_of_obs <- trunc_rep_tri_list[1:n]
+  # Truncate to only n nowcasts and extract only non-null elements of both lists
+  non_null_indices <- which(!sapply(pt_nowcast_mat_list[1:n], is.null))
+  n_iters <- length(non_null_indices)
+  list_of_ncs <- pt_nowcast_mat_list[non_null_indices]
+  list_of_obs <- trunc_rep_tri_list[non_null_indices]
+  if (n_iters == 0) {
+    cli_abort(
+      message = c(
+        "No retrospective nowcasts were generated, therefore ",
```

Comment:
```suggestion
        "No valid retrospective nowcasts were found, therefore ",
```

they are an input not generated here.